.github/workflows/lock-dependency.yml

```diff
@@ -91,7 +91,7 @@ jobs:
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          persist-credentials: true
+          persist-credentials: true # `persist-credentials: true` is required for `git` operations
```

Comment:
not quite helpful, this is only a describing comment ("of course" it is for git operations because that thing is persisting git auth on a git clone)

I presume this is because of git-auto-commit-action, thus something like that:

```
persist-credentials: true # required for `git-autocommit-action` because XYZ
```

Where `XYZ` is actually the important bit.

It might even hint at how to do it (if it can be done) without persisting; e.g maybe the secret token needs to be passed to the action via some `with: secrets:`?

Reply:
https://github.com/DataDog/dd-trace-rb/pull/4491